### ISSUE TITLE
node_set_size of StructuraMLP doesn't work in MC-SVD for link prediction.

### ISSUE THREAD

**Issue by LiamMa** [bala-srini/Equivalence#1]
Hi, 

Thank you for sharing codes for such an excellent work.

However, I encounter some problems with the code of MC-SVD for link prediction.
The argument of ```node_set_size``` seems like not changing anything inside ```StructuralMLP``` no matter in forward or initialization. 
It is supposed to be essential in getting Joint structural representation from node embeddings according to the paper.

Is there anything missing in this part of the code?

Thank you in advance.

**Comment by balasrini32** [comment]
Hi,

Thanks for the interest.
Great catch. The variable is unused as the node set size is directly inferred in the deep sets forward path - where the latent representations of the all the nodes in the node set is summed up.
i.e. 
@ x = torch.sum(x, dim=1)

Also, please ignore the comment:
#Process the input tensor to form n choose k combinations and create a zero tensor

We will fix the code to remove the unused variables.

**Comment by LiamMa** [comment]
Thank you for the quick response.
It clears up my confusion.

**Comment by balasrini32** [comment]
Thanks. I have closed the issue. Feel free to reopen it if you have more queries - will be happy to answer them
